### PR TITLE
applications: machine_learning: Add sensor_sim_ctrl

### PR DESCRIPTION
**Pull request by MarekPieta**
Change adds module to control generated sensor signal.

Jira: NCSDK-8457

### CODE REVIEW

File: applications/machine_learning/src/modules/Kconfig.sensor_sim_ctrl [hunk 1]
```diff
@@ -0,0 +1,49 @@
+#
+# Copyright (c) 2021 Nordic Semiconductor
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+
+menu "Sensor sim controller"
```
Comment (pdunaj) [inline applications/machine_learning/src/modules/Kconfig.sensor_sim_ctrl:7]: Simulated sensor controller

**Review by pdunaj — CHANGES_REQUESTED** [review]

File: applications/machine_learning/src/modules/sensor_sim_ctrl.c [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+	if ((event->key_id == WAVE_SWAP_BUTTON_ID) && event->pressed) {
+		select_next_wave();
+	}
+
+	return false;
+}
+
+static void verify_wave_params(void)
+{
+	const struct wave_gen_param *w;
```
[the remark this answers is not in the document]
Comment (MarekPieta) [reply applications/machine_learning/src/modules/sensor_sim_ctrl.c:93]: done

File: applications/machine_learning/src/modules/Kconfig.sensor_sim_ctrl (hunk 1 above)
Comment (MarekPieta) [reply applications/machine_learning/src/modules/Kconfig.sensor_sim_ctrl:7]: done

**Review by pdunaj — APPROVED** [review]

**Review by emob-nordic — APPROVED** [review]
Remember to add documentation for that later on. The comments are optional from my pov. If you want to make caf-like Kconfigs follow the comments, if no skip it.

File: applications/machine_learning/src/modules/Kconfig.sensor_sim_ctrl [hunk 3]
```diff
@@ -0,0 +1,49 @@
+#
+# Copyright (c) 2021 Nordic Semiconductor
```
[the remark this answers is not in the document]
Comment (MarekPieta) [reply applications/machine_learning/src/modules/Kconfig.sensor_sim_ctrl:2]: done

File: applications/machine_learning/src/modules/Kconfig.sensor_sim_ctrl [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+
+menu "Simulated sensor controller"
+
+config ML_APP_SENSOR_SIM_CTRL_ENABLE
+	bool "Enable simulated sensor controller"
```
[the remark this answers is not in the document]
Comment (MarekPieta) [reply applications/machine_learning/src/modules/Kconfig.sensor_sim_ctrl:10]: done

**Pull request merged**